cpp/include/cudf/io/parquet_schema.hpp

```diff
@@ -27,10 +25,143 @@
+                    // as an INT64 physical type.
+  TIMESTAMP_MILLIS = 9,   // A date/time combination, recorded as milliseconds since the Unix epoch
+                          // using physical type of INT64.
+  TIMESTAMP_MICROS = 10,  // A date/time combination, microseconds since the Unix epoch as INT64
+  UINT_8           = 11,  // An unsigned integer 8-bit value as INT32
+  UINT_16          = 12,  // An unsigned integer 16-bit value as INT32
+  UINT_32          = 13,  // An unsigned integer 32-bit value as INT32
+  UINT_64          = 14,  // An unsigned integer 64-bit value as INT64
+  INT_8            = 15,  // A signed integer 8-bit value as INT32
+  INT_16           = 16,  // A signed integer 16-bit value as INT32
+  INT_32           = 17,  // A signed integer 32-bit value as INT32
+  INT_64           = 18,  // A signed integer 8-bit value as INT64
+  JSON             = 19,  // A JSON document embedded within a single UTF8 column.
+  BSON             = 20,  // A BSON document embedded within a single BINARY column.
+  INTERVAL = 21,  // This type annotates a time interval stored as a FIXED_LEN_BYTE_ARRAY of length
+                  // 12 for 3 integers {months,days,milliseconds}
+  NA = 25,        // No Type information, For eg, all-nulls.
+};
+
+/**
+ * @brief Encoding types for the actual data stream
+ */
+enum class Encoding : uint8_t {
+  PLAIN                   = 0,
+  GROUP_VAR_INT           = 1,  // Deprecated, never used
+  PLAIN_DICTIONARY        = 2,
+  RLE                     = 3,
+  BIT_PACKED              = 4,  // Deprecated by parquet-format in 2013, superseded by RLE
+  DELTA_BINARY_PACKED     = 5,
+  DELTA_LENGTH_BYTE_ARRAY = 6,
+  DELTA_BYTE_ARRAY        = 7,
+  RLE_DICTIONARY          = 8,
+  BYTE_STREAM_SPLIT       = 9,
+  NUM_ENCODINGS           = 10,
+};
+
+/**
+ * @brief Compression codec used for compressed data pages
+ */
+enum Compression {
+  UNCOMPRESSED = 0,
+  SNAPPY       = 1,
+  GZIP         = 2,
+  LZO          = 3,
+  BROTLI       = 4,  // Added in 2.3.2
+  LZ4          = 5,  // deprecated; based on LZ4, but with an additional undocumented framing scheme
+  ZSTD         = 6,  // Added in 2.3.2
+  LZ4_RAW      = 7,  // "standard" LZ4 block format
+};
+
+/**
+ * @brief Compression codec used for compressed data pages
+ */
+enum FieldRepetitionType {
+  NO_REPETITION_TYPE = -1,
+  REQUIRED = 0,  // This field is required (can not be null) and each record has exactly 1 value.
+  OPTIONAL = 1,  // The field is optional (can be null) and each record has 0 or 1 values.
+  REPEATED = 2,  // The field is repeated and can contain 0 or more values
+};
+
+/**
+ * @brief Types of pages
+ */
+enum class PageType : uint8_t {
+  DATA_PAGE       = 0,
+  INDEX_PAGE      = 1,
+  DICTIONARY_PAGE = 2,
+  DATA_PAGE_V2    = 3,
+};
+
+/**
+ * @brief Enum to annotate whether lists of min/max elements inside ColumnIndex
+ * are ordered and if so, in which direction.
+ */
+enum BoundaryOrder {
```

Comment:
should we make every enum an enum class? making such changes later would be breaking

Reply:
Yeah not a bad idea imho

Reply:
Done in #18441

Reply:
casual 500 lines
if anyone asks, it had to be done ;)